### PR TITLE
SAK-29814 check for ToolsPlacement value in order to avoid the NPE error

### PR DESCRIPTION
**Pull request by zqian**
Place null value check for Tool placement, which is null for entityprovider requests.

### CODE REVIEW

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java [hunk 1]
```diff
@@ -3125,10 +3125,13 @@ else if(contentService.isPubView(resource.getId()))
 		{
 			siteId = ref.getContext();
 		}
-		else
+		
+		// id may be null in format of /group/<site_id>, which leads to null value for the reference context field
+		if (siteId == null && ToolManager.getCurrentPlacement() != null)
```
Comment (pushyamig) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java]: What if this condition is not true then the siteId will be null, is that OK?

**Comment by lsloan** [comment]
Is this URL from the JIRA a test to prove your changes are working?

https://trunk-mysql.nightly.sakaiproject.org/direct/content/site/mercury.json

Did it fail before your changes?

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java (hunk 1 above)
Comment (lsloan) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java]: To put it more specifically, what happens when both `siteId ` and `ToolManager.getCurrentPlacement()` ***are*** null?  Will calls to `SiteService.allowUpdateSite()` fail?

**Comment by zqian** [comment]
The above url works when there is no content (collection nor resource item) within the site Resources tool. Once you've added content in there, you would see the reported error with the above url.

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java (hunk 1 above)
Comment (zqian) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java]: null value of siteId won't make SiteService.allowUpdateSite() call fail. Nevertheless, I added a routine there to get the site id from collection reference string.

**Comment by zqian** [comment]
merge after code review

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java [hunk 2]
```diff
@@ -3140,6 +3140,26 @@ else if(contentService.isPubView(resource.getId()))
...
+		
+		if (siteId == null)
+		{
+			// if the siteId is still null
+			//// find the site root collection id			
+			org.sakaiproject.content.api.ContentHostingService contentService = ContentHostingService.getInstance();
+			String collectionId = id;
+			String containingCollectionId = ContentHostingService.getContainingCollectionId(id);
+			// containingCollectionId becomes "/group/" when checking against the root level collection
+			while (!contentService.COLLECTION_SITE.equals(containingCollectionId))
```
Comment (bkirschn) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java]: Endless loop :-(

**Pull request merged**